### PR TITLE
Iterate over the protocols to pick a working one.

### PR DESCRIPTION
**Pull request by vifino**
Hello.

Very nice project you have going here - Had the same idea, but didn't get around to doing it.
Had a look at your structure, very nice!

While going through your code, I found a piece of code where I would place a loop for several reasons, so here's a PR!

Pretty sure I followed the code style as well.
Not like it's a big change, but at least I get to say thank you for your work. ;)

### CODE REVIEW

**Comment by vifino** [comment]
Note: I did not update the minified version, there is no minifier included in this project, not by code or by name, as far as I can tell.
I don't use the minified version either, I just enabled luac in the firmware and everything's fine.

**Comment by vifino** [comment]
Hold on, I fucked up, give me a second.

**Comment by codecae** [comment]
I can minify it

File: src/SCRIPTS/BF/protocols.lua [hunk 1]
```diff
@@ -25,10 +25,11 @@ supportedProtocols =
 }
 
 function getProtocol()
-    if supportedProtocols.smartPort.push() then
-        return supportedProtocols.smartPort
-    elseif supportedProtocols.crsf.push() then
-        return supportedProtocols.crsf
+    for _, protocol in pairs(supportedProtocols) do
+        local success, result = pcall(protocol.push)
```
Comment (codecae) [inline src/SCRIPTS/BF/protocols.lua]: The procedure call will always be successful no matter which protocol is being used.  Instead it should be tested for true/false.

Comment (codecae) [reply src/SCRIPTS/BF/protocols.lua]: https://opentx.gitbooks.io/opentx-2-2-lua-reference-guide/content/general/crossfireTelemetryPush.html

https://opentx.gitbooks.io/opentx-2-2-lua-reference-guide/content/general/sportTelemetryPush.html

Comment (codecae) [inline src/SCRIPTS/BF/protocols.lua]: ```if protocol.push()``` would work best in this scenario, because neither call will end in error.

Comment (vifino) [reply src/SCRIPTS/BF/protocols.lua]: I tried it that way, but it errored out saying that the telemetry protocol was not supported.

File: src/SCRIPTS/BF/protocols.lua [hunk 2]
```diff
@@ -25,10 +25,11 @@ supportedProtocols =
...
+        if success and result then
```
Comment (codecae) [inline src/SCRIPTS/BF/protocols.lua]: The pcall above would work if you simply verify the ```result``` here

File: src/SCRIPTS/BF/protocols.lua (hunk 1 above)
Comment (codecae) [reply src/SCRIPTS/BF/protocols.lua]: Were you testing on a transmitter or in a simulator?

File: src/SCRIPTS/BF/protocols.lua (hunk 2 above)
Comment (vifino) [reply src/SCRIPTS/BF/protocols.lua]: `result` would be the error message if it does fail, a `string`, which is truthy.

File: src/SCRIPTS/BF/protocols.lua (hunk 1 above)
Comment (vifino) [reply src/SCRIPTS/BF/protocols.lua]: I'll try to revert back on my radio, maybe it was just a weird, random error.

Comment (codecae) [reply src/SCRIPTS/BF/protocols.lua]: The reason I ask is I don't think the telemetry buffer will ever result to ```true``` upon test in a simulator, hence a protocol would not be determined in the block.

**Comment by codecae** [comment]
By the way -- Thank you for contributing :)   I really appreciate the interest of others in improving the code!  :+1:

**Comment by vifino** [comment]
Yup, it does seem that the pcall was not needed, still weird that it raised an error without, stating that the telemetry protocol is not supported... I had telemetry from my quad at that point.

**Comment by vifino** [comment]
Welp, I wonder what caused said message to appear. Might just be because I didn't remove all the luac files, just the protocols.luac file... But that doesn't make much sense. I have no idea.

It might not be a bad idea to keep the `pcall`, however, since we don't directly call the telemetry pusher functions. It doesn't hurt anything anyways. If we had a more complex telemetry pushing function, it might be failing more arbitrarily if it isn't supported.

Do you want me to revert it back to my original version, without the `pcall`? I'm guessing so. Give me a second, I'll try to convince git to let me.

**Comment by vifino** [comment]
Updated. Seems to work fine, after I cleaned up all the luac files.

File: src/SCRIPTS/BF/protocols.lua (hunk 1 above)
Comment (vifino) [reply src/SCRIPTS/BF/protocols.lua]: Didn't test in the simulator, I tested on my X9D+.

**Comment by codecae** [comment]
It's no trouble :)  I'm glad you want to help

**Comment by codecae** [comment]
Personally I think the original commit you had was simpler and accomplished the same goal.

**Comment by codecae** [comment]
I have some other stuff I'm working on if you'd like to lend a hand.

**Comment by vifino** [comment]
Sure! ;)

Don't have a crossfire, unfortunately, so I can only test smart port telemetry.

**Comment by codecae** [comment]
no matter.. should apply either way. I'll open an enhancement issue and invite you to the discussion

**Comment by vifino** [comment]
Sounds good.

**Pull request merged**